rpcclient/infrastructure.go

```diff
@@ -924,13 +931,60 @@ func (c *Client) nextPostRequest() chan *jsonRequest {
 // while allowing the sender to continue running asynchronously.  It must be run
 // as a goroutine.
 func (c *Client) sendPostHandler() {
+	// Add an errgroup to handle the post messages in goroutines.
+	var eg errgroup.Group
+
+	// Limit the number of concurrent RPC calls.
+	//
+	// TODO(yy): Ideally we should pass the `-rpcthreads` used in
+	// `bitcoind` here to better increase the performance.
+	const numThreads = 4
+	eg.SetLimit(numThreads)
```

Comment:
Don't we ever need to wait for this error group? Or we're basically using it as a simple "thread pool"?

Reply:
yeah it's more like limiting concurrent calls, but think we should also wait in the end so now I added it.